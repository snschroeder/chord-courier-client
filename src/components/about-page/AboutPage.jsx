import React from 'react';
import LightNav from '../Nav/LightNav';
import LightHeader from '../Header/LightHeader';
import './about-page.css';

export default function AboutPage() {
  return (
    <>
      <LightNav />
      <section className="about-page">
        <div className="about-image">
          <div className="about-text">
            <LightHeader />
            <h3 className="about-h3">your catalyst for musical inspiration</h3>
            <p className="about-p">
              Chord Courier allows you to explore guitar chords, generating common four chord
               progressions that you can customize from the chord gallery to make your own.
               Use the progression randomizer as a launch point for your songwriting and musical
               discovery.
            </p>
            <h4 className="about-h4">Explore chord by key</h4>
            <p className="about-p">
              Feel free to search through all chords in a given key to build your own progressions.
               Take some time to expand your chord knowledge.
            </p>
            <h4 className="about-h5">Over 250 chords</h4>
            <p className="about-p">
              You will find everything from the basics to chords right at home in jazz,
               rock, pop or indie songs.
            </p>
          </div>
        </div>
      </section>
    </>
  );
}
