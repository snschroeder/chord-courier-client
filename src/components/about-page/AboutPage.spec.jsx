/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AboutPage from './AboutPage';

describe('AboutPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AboutPage />
      </BrowserRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AboutPage />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
