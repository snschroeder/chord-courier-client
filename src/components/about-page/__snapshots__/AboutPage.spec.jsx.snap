// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage component renders the UI as expected 1`] = `
Array [
  <nav
    className="nav-bar"
  >
    <ul
      className="link-list"
    >
      <a
        href="/search"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li
          className="nav-link"
        >
          Progression Randomizer
        </li>
      </a>
      <a
        href="/full-search"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li
          className="nav-link"
        >
          Chords by Key
        </li>
      </a>
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li
          className="nav-link"
        >
          About
        </li>
      </a>
    </ul>
  </nav>,
  <section
    className="about-page"
  >
    <div
      className="about-image"
    >
      <div
        className="about-text"
      >
        <section>
          <h1
            className="light-header-h1"
          >
            Chord Courier
          </h1>
          <h2
            className="light-header-h2"
          >
            Delivering fresh chords
          </h2>
        </section>
        <h3
          className="about-h3"
        >
          your catalyst for musical inspiration
        </h3>
        <p
          className="about-p"
        >
          Chord Courier allows you to explore guitar chords, generating common four chord progressions that you can customize from the chord gallery to make your own. Use the progression randomizer as a launch point for your songwriting and musical discovery.
        </p>
        <h4
          className="about-h4"
        >
          Explore chord by key
        </h4>
        <p
          className="about-p"
        >
          Feel free to search through all chords in a given key to build your own progressions. Take some time to expand your chord knowledge.
        </p>
        <h4
          className="about-h5"
        >
          More chords coming
        </h4>
        <p
          className="about-p"
        >
          More chords will be added as time allows...
        </p>
      </div>
    </div>
  </section>,
]
`;
